### ISSUE TITLE
FMT003: Detect counter syntax in section titles

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#29]
## Problem

LLMs sometimes generate section titles with AsciiDoc counter syntax:

```asciidoc
== Phase {counter:phase}
== Step {counter:step}: Setup
== Section {counter:section}
```

This pattern is often unnecessary or misused because:
- Simple sequential sections don't need explicit counters
- Counter state can be confusing across includes
- Often the LLM is trying to mimic numbered headings from other formats

## Proposed Solution

Add a new rule `FMT003` that detects `{counter:...}` usage in section titles and warns about it.

The rule should:
1. Detect `{counter:name}` pattern in lines starting with `=`
2. Suggest reviewing if counter is really needed
3. Optionally suggest using natural language or removing counter

## Examples

**Should trigger warning:**
```asciidoc
== Phase {counter:phase}
=== Step {counter:step}: Configuration
```

**Alternatives to suggest:**
```asciidoc
== Phase 1: Discovery
== Phase 2: Implementation
```
or simply:
```asciidoc
== Discovery Phase
== Implementation Phase
```

## Acceptance Criteria
- [ ] New rule `FMT003` in `format_rules.py`
- [ ] Detects `{counter:...}` in section titles
- [ ] Provides helpful suggestion
- [ ] Tests cover various counter patterns
